apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts

```diff
@@ -149,15 +148,22 @@ export class ExpandEmailEditorSchemaUsecase {
    * ]
    *
    */
-  private multiplyForEachNode(
+  private async multiplyForEachNode(
     node: TipTapNode & { attrs: { [MailyAttrsEnum.EACH_KEY]: string } },
     variables: FullPayloadForRender
-  ): TipTapNode[] {
+  ): Promise<TipTapNode[]> {
     const iterablePath = node.attrs[MailyAttrsEnum.EACH_KEY];
     const nodeContent = node.content || [];
     const expandedContent: TipTapNode[] = [];
 
-    const iterableArray = this.getValueByPath(variables, iterablePath);
+    const iterableArrayString = await parseLiquid(iterablePath, variables);
+
+    let iterableArray: unknown;
+    try {
+      iterableArray = JSON.parse(iterableArrayString.replace(/'/g, '"'));
+    } catch (error) {
+      throw new Error(`Failed to parse iterable value for "${iterablePath}": ${error.message}`);
+    }
```

Comment:
safeguard since we are calling `JSON.parse()` on something that is output from liquid, which parses something that user provides etc